### ISSUE TITLE
`toggleShareScreen` command stopped working

### ISSUE THREAD

**Issue by geiszla** [jitsi/jitsi-meet#11916]
### Description:

When executing the `toggleShareScreen` command of the IFrame API (using `meet.jit.si` as the server), nothing happens. Previously (a few days ago) it was working fine.

### Steps to reproduce:

1. `const meetingApi = new JitsiMeetExternalAPI('meet.jit.si', apiOptions)`
1. Create a button and add `() => meetingApi.executeCommand('toggleShareScreen')` as the `onClick` listener to it.
2. Click the button

### Expected behavior:

Screen share modal should open

### Actual behavior:

Nothing happens

### Server information:

- Jitsi Meet version: latest version on meet.jit.si
- Operating System: whatever operating system meet.jit.si uses

### Client information:

- Browser / app version: Google Chrome/Firefox latest (probably others as well)
- Operating System: Windows/MacOs (probably Linux as well)

**Comment by saghul** [comment]
Starting a screen share now requires a user gesture, aka a button press. This is something browsers have changed and there is nothing we can do about it.

@jallamsetty1 should we document this in the iframe API perhaps?

**Comment by geiszla** [comment]
We put this function on a button (see "Steps to reproduce" above), so there is a user interation before it runs. Or what do you mean exactly?

**Comment by saghul** [comment]
The origin of the event won't traverse the iframe boundary. Your button is your site, but the call to `getDisplayMedia` happens inside the iframe because you send the command, so the source of the call to `getDisplayMedia` is lost. As such, the browser thinks you tried to call it programmatically and it will fail.

**Comment by geiszla** [comment]
If this was the case, wouldn't Chrome (or any other browser) display a warning or error (like it does for `.play()` when it's not called on user interaction)? Also, it is very strange that it stopped working in all browsers (Chrome/Firefox) the same time. What are the chances that they release a change like this and all users update it at the same time?

**Comment by saghul** [comment]
Hum, looks like it should work in Chrome at least indeed, since they haven't solved that yet: https://bugs.chromium.org/p/chromium/issues/detail?id=1198918

Note that this doesn't work on Safari or Firefox and Chrome will be coming, so I'm not sure if the effort to fix the bug is warranted.

**Comment by geiszla** [comment]
If it's not too much effort, can you take a look into it until we have time to adjust? We're also testing with Chrome Dev to have early warnings of these behaviour changes, but as this is not implemented yet, we had no warning that this would become unavailable. This feature is quite central to our app and would need some restructuring of the UI to solve (since all our buttons to Jitsi are custom, we either need to completely change the UI to use Jitsi buttons or move this feature inside our app, which takes some time either way).

Btw we were also using it from Safari and it was working about a week ago. We haven't tested it directly on Firefox, but ~100 of our monthly users use Firefox and we didn't have any bug reports about this before. I'm not sure that these are the latest versions of the mentioned browsers, but considering that this issue haven't come up before we can safely say that most people use browsers unaffected by this change.

**Comment by geiszla** [comment]
Update: we found that using `meetingApi.executeCommand('toggleShareScreen', { enable: true })` (with an added `enable: true` option) works on browsers, where the user interaction restrictions have not been implemented yet, however the `screenSharingStatusChanged` event doesn't fire. This event has nothing to do with the ability to send commands without user interaction, so I expect this will be fixed?

It probably would also worth updating the docs to correctly reflect the usage of the `toggleShareScreen` command.

**Comment by saghul** [comment]
Thanks for looking into it @jallamsetty1 !

**Comment by geiszla** [comment]
@jallamsetty1 Thank for you help on this issue. Could it be that the multi-stream changes broke the camera feeds in the mobile SDK? Recently our mobile app broke (without us changing anything) in a way that we can't see any camera feed from any other user and the UI shows that their camera is disabled (in reality, they aren't). We are using `org.jitsi.react:jitsi-meet-sdk:4.1.0`, because that's the last version we could upgrade to that was still supported by the React Native wrapper (the official Flutter/React Native repos still use `v3.x`). Is this a known problem? Should I create a new issue for this?

**Comment by saghul** [comment]
Can you please open a new issue about that, since it's not related to the iframe API but to the mobile SDKs?

**Comment by geiszla** [comment]
https://github.com/jitsi/jitsi-meet/issues/11977
